### PR TITLE
[feat] Added Jigsaw Augmentation

### PR DESCRIPTION
**Pull request by shijianjian**
Jigsaw augmentation
image

### CODE REVIEW

**Comment by edgarriba** [comment]
@shijianjian whats the difference between the mosaic ?

**Comment by shijianjian** [comment]
> @shijianjian whats the difference between the mosaic ?

Jigsaw mixes single images only. Mosaic interpolates multiples.

**Comment by shijianjian** [comment]
To add, this implementation is differentiable though the gradients are flowed as using a STE, while the mosaic is currently not.

**Comment by shijianjian** [comment]
@edgarriba  review and merge

File: kornia/augmentation/_2d/mix/jigsaw.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+__all__ = ["RandomJigsaw"]
+
+
+class RandomJigsaw(MixAugmentationBaseV2):
+    r"""RandomJigsaw augmentation.
+
+    .. image:: https://raw.githubusercontent.com/kornia/data/main/random_jigsaw.png
+
+    Make Jigsaw puzzles for each image individually.
```
Comment (edgarriba) [inline kornia/augmentation/_2d/mix/jigsaw.py:23]: maybe add note to distinguish with random mosaic -- it can be easily confused

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@shijianjian fix code format and good to go

File: kornia/augmentation/random_generator/_2d/jigsaw.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import torch
+
+from kornia.augmentation.random_generator.base import RandomGeneratorBase
+from kornia.augmentation.random_generator.utils import randperm
+from kornia.augmentation.utils import _common_param_check
+
+
+class JigsawGenerator(RandomGeneratorBase):
+    r"""Generate Jigsaw permutation indicies for a batch of inputs.
```
Comment (shijianjian) [inline kornia/augmentation/random_generator/_2d/jigsaw.py:11]:
```suggestion
    r"""Generate Jigsaw permutation indices for a batch of inputs.
```

**Review by edgarriba — COMMENTED** [review]
amazing ! can you add or open ticket so that we can have a tutorial or something ?

**Comment by Twsaiae** [comment]
I have a question, which is how do the category information of labels correspond when using Mosaic enhancement on object detection datasets? What is the category corresponding to each label in the generated graph?

**Comment by edgarriba** [comment]
@shijianjian any thoughts ? Haven’t tried personally this augmentation yet
